src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -31,7 +32,33 @@ public List<BookDto> findAll() {
 
     @Override
     public BookDto getBookById(Long id) {
-        Book bookById = bookRepository.getBookById(id);
-        return bookMapper.toDto(bookById);
+        return bookRepository.findById(id).stream()
+                .map(bookMapper::toDto)
+                .findFirst()
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "can`t find book with id: " + id));
+    }
+
+    @Override
+    public BookDto updateBookById(Long id, CreateBookRequestDto updatedBookDto) {
+        Book existingBook = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Book not found with id: " + id));
+
+        if (updatedBookDto == null) {
+            throw new EntityNotFoundException("please provide update details ");
+        }
+        existingBook.setTitle(updatedBookDto.getTitle());
+        existingBook.setAuthor(updatedBookDto.getAuthor());
+        existingBook.setDescription(updatedBookDto.getDescription());
+        existingBook.setIsbn(updatedBookDto.getIsbn());
+        existingBook.setPrice(updatedBookDto.getPrice());
+        existingBook.setCoverImage(updatedBookDto.getCoverImage());
+        return bookMapper.toDto(bookRepository.save(existingBook));
```

Comment:
it can be a bit simpler than that:
```suggestion
        var updatedBook = bookMapper.toModel(updatedBookDto);
        updatedBook.setId(id);
        return repo.save(updatedBook);
```
because hibernate will know that we need to update the fields of the existing book by the fact that we have set an id of an already existing book